### ISSUE TITLE
Unable to load Preflight page Windows Server 2016

### ISSUE THREAD

**Issue by smannon** [snipe/snipe-it#6605]
#### Expected Behavior (or desired behavior if a feature request)

After following all the installation steps and configuration of MariaDB as well as the .env file and IIS configuration, I expected to be taken to the "preflight" page.

-----

#### Actual Behavior

Whoops something went wrong....

(1/1)                                     InvalidArgumentException                                 Database [Snipe-IT] not configured.
--
in DatabaseManager.php line 140
at DatabaseManager->configuration('Snipe-IT')in DatabaseManager.php line 103
at DatabaseManager->makeConnection('Snipe-IT')in DatabaseManager.php line 74
at DatabaseManager->connection()in Schema.php line 33
at Schema::getFacadeAccessor()in Facade.php line 128
at Facade::getFacadeRoot()in Facade.php line 215
at Facade::__callStatic('defaultStringLength', array(191))in AppServiceProvider.php line 38
at AppServiceProvider->boot()
at call_user_func_array(array(object(AppServiceProvider), 'boot'), array())in BoundMethod.php line 29
at BoundMethod::Illuminate\Container\{closure}()in BoundMethod.php line 87
at BoundMethod::callBoundMethod(object(Application), array(object(AppServiceProvider), 'boot'), object(Closure))in BoundMethod.php line 31
at BoundMethod::call(object(Application), array(object(AppServiceProvider), 'boot'), array(), null)in Container.php line 539
at Container->call(array(object(AppServiceProvider), 'boot'))in Application.php line 788
at Application->bootProvider(object(AppServiceProvider))in Application.php line 771
at Application->Illuminate\Foundation\{closure}(object(AppServiceProvider), 22)
at array_walk(array(object(EventServiceProvider), object(LogServiceProvider), object(RoutingServiceProvider), object(AuthServiceProvider), object(CookieServiceProvider), object(DatabaseServiceProvider), object(EncryptionServiceProvider), object(FilesystemServiceProvider), object(FormRequestServiceProvider), object(FoundationServiceProvider), object(NotificationServiceProvider), object(PaginationServiceProvider), object(SessionServiceProvider), object(ViewServiceProvider), object(ServiceProvider), object(ImageServiceProvider), object(BackupServiceProvider), object(TrustedProxyServiceProvider), object(PassportServiceProvider), object(DumbPasswordServiceProvider), object(PasswordStrengthServiceProvider), object(ZiggyServiceProvider), object(AppServiceProvider), object(AuthServiceProvider), object(EventServiceProvider), object(RouteServiceProvider), object(SettingsServiceProvider), object(ValidationServiceProvider), object(MacroServiceProvider), object(MailServiceProvider), object(QueueServiceProvider), object(ValidationServiceProvider), object(TranslationServiceProvider)), object(Closure))in Application.php line 772
at Application->boot()in BootProviders.php line 17
at BootProviders->bootstrap(object(Application))in Application.php line 208
at Application->bootstrapWith(array('Illuminate\\Foundation\\Bootstrap\\LoadEnvironmentVariables', 'Illuminate\\Foundation\\Bootstrap\\LoadConfiguration', 'Illuminate\\Foundation\\Bootstrap\\HandleExceptions', 'Illuminate\\Foundation\\Bootstrap\\RegisterFacades', 'Illuminate\\Foundation\\Bootstrap\\RegisterProviders', 'Illuminate\\Foundation\\Bootstrap\\BootProviders'))in Kernel.php line 162
at Kernel->bootstrap()in Kernel.php line 146
at Kernel->sendRequestThroughRouter(object(Request))in Kernel.php line 116
at Kernel->handle(object(Request))in index.php line 58

-----

#### Please confirm you have done the following before posting your bug report:

- [x] I have enabled debug mode 
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Provide answers to these questions:

- Is this a fresh install or an upgrade? - Fresh install
- Version of Snipe-IT you're running - 4.6.7
- Version of PHP you're running - 7.3.1
- Version of MySQL/MariaDB you're running - 10.0.37
- What OS and web server you're running Snipe-IT on - Windows Server 2016
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) - Manual
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error - See above
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error - http://localhost
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console. - None
- Confirm whether the error is reproducible on the demo: https://snipeitapp.com/demo.  - No
- Include any additional information you can find in `storage/logs` and your webserver's logs. 
- Include what you've done so far in the installation, and if you got any error messages along the way. - No errors along the way
- Indicate whether or not you've manually edited any data directly in the database - User Name and Password

Please do not post an issue without answering the related questions above. If you have opened a different issue and already answered these questions, answer them again, once for every ticket. It will be next to impossible for us to help you.

https://snipe-it.readme.io/docs/getting-help

**Comment by snipe** [comment]
Can you post your env (redacting passwords). It looks like you’ve got the wrong value for something in the database section in there.

**Comment by smannon** [comment]
# --------------------------------------------
# REQUIRED: BASIC APP SETTINGS
# --------------------------------------------
APP_ENV=production
APP_DEBUG=true
APP_KEY=base64:r564q7pVhLFjkjTdcT/y/nerFOEFAsPAdCw2qXBEw2o=
APP_URL=http://snipeit
APP_TIMEZONE='UTC'
APP_LOCALE=en

# --------------------------------------------
# REQUIRED: DATABASE SETTINGS
# --------------------------------------------
DB_CONNECTION=snipe-it
DB_HOST=127.0.0.1
DB_DATABASE=snipe-it
DB_USERNAME=snipeit
DB_PASSWORD=XXXXXXXXXX
DB_PREFIX=null
DB_DUMP_PATH='C:\Program Files\MariaDB 10.0\bin'
DB_CHARSET=utf8mb4
DB_COLLATION=utf8mb4_unicode_ci

# --------------------------------------------
# OPTIONAL: SSL DATABASE SETTINGS
# --------------------------------------------
DB_SSL=false
DB_SSL_KEY_PATH=null
DB_SSL_CERT_PATH=null
DB_SSL_CA_PATH=null
DB_SSL_CIPHER=null

# --------------------------------------------
# REQUIRED: OUTGOING MAIL SERVER SETTINGS
# --------------------------------------------
MAIL_DRIVER=smtp
MAIL_HOST=email-smtp.us-west-2.amazonaws.com
MAIL_PORT=587
MAIL_USERNAME=YOURUSERNAME
MAIL_PASSWORD=YOURPASSWORD
MAIL_ENCRYPTION=null
MAIL_FROM_ADDR=you@example.com
MAIL_FROM_NAME='Snipe-IT'
MAIL_REPLYTO_ADDR=you@example.com
MAIL_REPLYTO_NAME='Snipe-IT'

# --------------------------------------------
# REQUIRED: IMAGE LIBRARY
# This should be gd or imagick
# --------------------------------------------
IMAGE_LIB=gd

# --------------------------------------------
# OPTIONAL: SESSION SETTINGS
# --------------------------------------------
SESSION_LIFETIME=12000
EXPIRE_ON_CLOSE=false
ENCRYPT=false
COOKIE_NAME=snipeit_session
COOKIE_DOMAIN=null
SECURE_COOKIES=false

# --------------------------------------------
# OPTIONAL: SECURITY HEADER SETTINGS
# --------------------------------------------
REFERRER_POLICY=same-origin
ENABLE_CSP=false

# --------------------------------------------
# OPTIONAL: CACHE SETTINGS
# --------------------------------------------
CACHE_DRIVER=file
SESSION_DRIVER=file
QUEUE_DRIVER=sync
CACHE_PREFIX=snipeit

# --------------------------------------------
# OPTIONAL: REDIS SETTINGS
# --------------------------------------------
REDIS_HOST=null
REDIS_PASSWORD=null
REDIS_PORT=null

# --------------------------------------------
# OPTIONAL: MEMCACHED SETTINGS
# --------------------------------------------
MEMCACHED_HOST=null
MEMCACHED_PORT=null

# --------------------------------------------
# OPTIONAL: AWS S3 SETTINGS
# --------------------------------------------
AWS_SECRET=null
AWS_KEY=null
AWS_REGION=null
AWS_BUCKET=null

# --------------------------------------------
# OPTIONAL: LOGIN THROTTLING
# --------------------------------------------
LOGIN_MAX_ATTEMPTS=5
LOGIN_LOCKOUT_DURATION=60

# --------------------------------------------
# OPTIONAL: MISC
# --------------------------------------------
APP_LOG=single
APP_LOG_MAX_FILES=10
APP_LOCKED=false
FILESYSTEM_DISK=local
APP_TRUSTED_PROXIES=192.168.1.1,10.0.0.1
ALLOW_IFRAMING=false
APP_CIPHER=AES-256-CBC
GOOGLE_MAPS_API=
BACKUP_ENV=true

**Comment by smannon** [comment]
Thanks for the support and the quick response.  I was off the last two days so this is the first opportunity I was able to get the information posted. 

Again thanks for the quick response.

**Comment by snipe** [comment]
Your DB_CONNECTION value is wrong.

It should be:

`DB_CONNECTION=mysql`

https://snipe-it.readme.io/docs/configuration#section-required-database-settings 

<img width="807" alt="screen shot 2019-01-24 at 1 01 44 pm" src="https://user-images.githubusercontent.com/197404/51708245-41de5e80-1fd8-11e9-80f7-cdb453c12c00.png">

**Comment by snipe** [comment]
Also you should generate a new app key, since yours was posted here. I redacted it, but the Internet never forgets.

Run `php artisan key:generate` to generate a fresh key.

**Comment by smannon** [comment]
We were able to figure that out.  Thanks of the help though.  I haven’t seen anything that outlines best practices at setting up the solution.  Is that anything that you can point me to?

I’ll close this request shortly.  Thanks again.

Next time off:   January 25th, 2019

Consider the destination of your choices before you commit to them, in order to prevent an  unacceptable outcome .  You life is full of choices so choose wisely, with meaning, and purposeful.

Steve Mannon – Information Systems Analyst II
 Information Services Analyst II |  San Joaquin General Hospital | 500 Hospital Road, French Camp CA 95231 | Phone: (209) 468-6282 | Fax: (209) 468-6546 |  Email: smannon@sjgh.org<mailto:smannon@sjgh.org>

**Comment by snipe** [comment]
> I haven’t seen anything that outlines best practices at setting up the solution

It's hard to say, since different organizations use pretty dramatically different workflows. Can you tell me more about how assets are currently given to users, whether you need to audit them, etc? How many users do you have? (The workflow can change significantly if you have 100 users versus 10,000).

**Comment by smannon** [comment]
Let me rephrase the question.
When setting up the Assets, users, Groups, Labels, locations, departments etc… Is there a best practice as to the order this is setup in?  It doesn’t seem like there is but I wanted to make sure.

By the way we have approx. 2500 users, with about 2000+ devices.

Next time off:   January 25th, 2019

Consider the destination of your choices before you commit to them, in order to prevent an  unacceptable outcome .  You life is full of choices so choose wisely, with meaning, and purposeful.

Steve Mannon – Information Systems Analyst II
 Information Services Analyst II |  San Joaquin General Hospital | 500 Hospital Road, French Camp CA 95231 | Phone: (209) 468-6282 | Fax: (209) 468-6546 |  Email: smannon@sjgh.org<mailto:smannon@sjgh.org>

**Comment by snipe** [comment]
That depends whether you have existing info that you can import, or whether you're starting from brand new with no existing asset information.

If you have existing asset information in a spreadsheet, you'll probably want to import that information first. The [importer documentation](https://snipe-it.readme.io/docs/importing) will walk you through what fields are required, etc. You may want to download one of the CSV templates and drop your data in, that way you know you've got everything covered. Importing will do a lot of the heavy lifting of creating users, manufacturers, models, etc. 

If you don't have existing data, you'll probably want to start with Categories or Asset Models. Once you have all of those outlined, you can create asset models on the fly as you manually create assets, so the workflow gets much easier. 

Asset models require categories, and although we have a modal window that lets you create asset models on the fly when creating assets, we don't have a second modal to allow you to create categories on the fly for the asset model you're trying to make on the fly. You can create categories on the fly when you're creating asset models normally though, so whether you start with categories or you start with asset models is up to you.

**Comment by smannon** [comment]
Perfect that is what I was looking for.  Thank you for your help!

Next time off:   January 25th, 2019

Consider the destination of your choices before you commit to them, in order to prevent an  unacceptable outcome .  You life is full of choices so choose wisely, with meaning, and purposeful.

Steve Mannon – Information Systems Analyst II
 Information Services Analyst II |  San Joaquin General Hospital | 500 Hospital Road, French Camp CA 95231 | Phone: (209) 468-6282 | Fax: (209) 468-6546 |  Email: smannon@sjgh.org<mailto:smannon@sjgh.org>
